packages/sit-onyx/src/composables/useCustomValidity.ts

```diff
@@ -22,6 +29,31 @@ export type UseCustomValidityOptions = {
 
 export type InputValidationElement = Pick<HTMLInputElement, "validity" | "setCustomValidity">;
 
+/**
+ * Input types that have a translation for their validation error message.
+ */
+export const TRANSLATED_INPUT_TYPES = Object.keys(
+  enUS.validations.typeMismatch,
+) as (keyof typeof enUS.validations.typeMismatch)[];
+export type TranslatedInputType = (typeof TRANSLATED_INPUT_TYPES)[number];
+
+/**
+ * Translated error messages that inform about causes for invalidity of form components
+ */
+export type FormErrorMessages =
+  | {
+      /**
+       * A short error message for giving a quick info to the user about the cause of the error
+       */
+      shortMessage: string;
+      /**
+       * An extended informative error message to provide more info
+       * how the error cause can be resolved
+       */
+      longMessage?: string;
+    }
+  | undefined;
```

Comment:
For the type itself I wouldn't add `| undefined` and instead add this where the type is used.
Otherwise the type could not be used "required"